### PR TITLE
Update PHPUnit to 10 and PHP to 8.1

### PR DESCRIPTION
**Pull request by gbirke**

### CODE REVIEW

File: tests/Unit/EuroTest.php [hunk 1]
```diff
@@ -10,62 +10,65 @@
...
 	 */
-	public function testGetCentsReturnsConstructorArgument( int $unsignedInteger ) {
+	public function testGetCentsReturnsConstructorArgument( int $unsignedInteger ): void {
 		$amount = Euro::newFromCents( $unsignedInteger );
 		$this->assertSame( $unsignedInteger, $amount->getEuroCents() );
 	}
 
-	public function unsignedIntegerProvider() {
+	/**
+	 * @return array{int}[]
```
Comment (JeroenDeDauw) [inline tests/Unit/EuroTest.php:25]: Interesting, never seen array shapes used like this. Probably `array<array<int>>` is clearer tho

Comment (JeroenDeDauw) [reply tests/Unit/EuroTest.php:25]: Or `int[][]`

Comment (gbirke) [reply tests/Unit/EuroTest.php:25]: `array{int}` is an [array shape](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes) that signifies a tuple with one `int` element, whereas `array<int>` is saying "this a list (arbitrary length) of integers" (same as `int[]`). `array{int}[]` means a list of these tuples.

In this case it doesn't matter (esp. for PHPStan), but where data providers return mixed tuples (e.g. `array{int,string}[]`) it is useful. 

I also like the precision of tuples that shows the exact number of elements in the array. Not in the language itself (yet), but at least as a PHPDdoc comment that the IDE and static analysis can interpret.

File: .gitignore [hunk 2]
```diff
@@ -5,8 +5,7 @@ composer.phar
 composer.lock
 
 .phpunit.result.cache
+.phpunit.cache
 
 vendor/
 
-.idea/
```
Comment (moiikana) [inline .gitignore:11]: what is the reason behind this change?

Comment (JeroenDeDauw) [reply .gitignore:11]: I have no .idea

Comment (gbirke) [reply .gitignore:11]: I think a code repository should not "depend" on any specific editor or IDE. Each developer has their own favourite editor that may leave artifacts in the folder. Ignoring those artifacts should be done in a [global gitignore](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c)

**Review by Abban — APPROVED** [review]
